managed/services/minio/client.go

```diff
@@ -0,0 +1,213 @@
+func (c *Client) BucketExists(ctx context.Context, endpoint, accessKey, secretKey, bucketName string) (bool, error) {
+	mc, err := createMinioClient(endpoint, accessKey, secretKey)
+	if err != nil {
+		return false, err
+	}
+	return mc.BucketExists(ctx, bucketName)
+}
+
+// GetBucketLocation retrieves bucket location by specified bucket name.
+func (c *Client) GetBucketLocation(ctx context.Context, endpoint, accessKey, secretKey, bucketName string) (string, error) {
+	mc, err := createMinioClient(endpoint, accessKey, secretKey)
+	if err != nil {
+		return "", err
+	}
+	return mc.GetBucketLocation(ctx, bucketName)
+}
+
+// RemoveRecursive removes objects recursively from storage with given prefix.
+func (c *Client) RemoveRecursive(ctx context.Context, endpoint, accessKey, secretKey, bucketName, prefix string) (rerr error) {
+	mc, err := createMinioClient(endpoint, accessKey, secretKey)
+	if err != nil {
+		return err
+	}
+
+	objectsCh := make(chan minio.ObjectInfo)
+	var g errgroup.Group
+	g.Go(func() error {
+		defer close(objectsCh)
+
+		options := minio.ListObjectsOptions{ //nolint:exhaustruct
+			Prefix:    prefix,
+			Recursive: true,
+		}
+		for object := range mc.ListObjects(ctx, bucketName, options) {
+			if object.Err != nil {
+				return errors.WithStack(object.Err)
+			}
+
+			objectsCh <- object
+		}
+
+		return nil
+	})
+
+	defer func() {
+		err := g.Wait()
+		if err == nil {
+			return
+		}
+
+		if rerr != nil {
+			rerr = errors.Wrapf(rerr, "listing objects error: %s", err.Error())
+		} else {
+			rerr = errors.WithStack(err)
+		}
+	}()
+
+	var errorsEncountered bool
+	for rErr := range mc.RemoveObjects(ctx, bucketName, objectsCh, minio.RemoveObjectsOptions{}) { //nolint:exhaustruct
+		errorsEncountered = true
+		c.l.WithError(rErr.Err).Debugf("failed to remove object %q", rErr.ObjectName)
+	}
+
+	if errorsEncountered {
+		return errors.Errorf("errors encountered while removing objects from bucket %q", bucketName)
+	}
+
+	return nil
+}
+
+// List provides an abstraction over the minio API to list all objects in the bucket
+// It scans path with prefix and returns all files with given suffix.
+// Both prefix and suffix can be omitted.
+func (c *Client) List(ctx context.Context, endpoint, accessKey, secretKey, bucketName, prefix, suffix string) ([]FileInfo, error) {
+	var files []FileInfo
+	if prefix != "" && !strings.HasSuffix(prefix, "/") {
+		prefix += "/"
+	}
+	mc, err := createMinioClient(endpoint, accessKey, secretKey)
+	if err != nil {
+		return nil, err
+	}
+
+	options := minio.ListObjectsOptions{
+		Prefix:    prefix,
+		Recursive: true,
+	}
+
+	for object := range mc.ListObjects(ctx, bucketName, options) {
+		if object.Err != nil {
+			return nil, errors.WithStack(object.Err)
+		}
+		f := object.Key
+		f = strings.TrimPrefix(f, options.Prefix)
+		if len(f) == 0 {
+			continue
+		}
+		if f[0] == '/' {
+			f = f[1:]
+		}
+
+		c.l.Infof(f)
```

Comment:
Let's remove this log, or make it `debug` and with more descriptive message.

Reply:
Let's also give a more clear name to var `f`)